modules/api/pkg/handler/v2/external_cluster/kubeone.go

```diff
@@ -86,131 +87,75 @@ func importKubeOneCluster(ctx context.Context, userInfoGetter func(ctx context.C
-	oldCluster *apiv2.ExternalCluster,
-	newCluster *apiv2.ExternalCluster,
+	existingCluster *kubermaticv1.ExternalCluster,
+	patchData json.RawMessage,
 	secretKeySelector provider.SecretKeySelectorValueFunc,
 	clusterProvider provider.ExternalClusterProvider,
 	masterClient ctrlruntimeclient.Client) (*apiv2.ExternalCluster, error) {
-	operation := cluster.Status.Condition.Phase
+	operation := existingCluster.Status.Condition.Phase
 	if operation == kubermaticv1.ExternalClusterPhaseReconciling {
 		return nil, utilerrors.NewBadRequest("Operation is not allowed: Another operation: (%s) is in progress, please wait for it to finish before starting a new operation.", operation)
 	}
 
-	if oldCluster.Spec.Version != newCluster.Spec.Version {
-		return UpgradeKubeOneCluster(ctx, cluster, oldCluster, newCluster, clusterProvider, masterClient)
-	}
-	if oldCluster.Spec.ContainerRuntime != newCluster.Spec.ContainerRuntime {
-		if oldCluster.Spec.ContainerRuntime == resources.ContainerRuntimeDocker {
-			return MigrateKubeOneToContainerd(ctx, cluster, oldCluster, newCluster, clusterProvider, masterClient)
-		} else {
-			return nil, fmt.Errorf("Operation not supported: only migration from docker to containerd is supported: %s", oldCluster.Spec.ContainerRuntime)
-		}
-	}
-
-	return newCluster, nil
-}
-
-func UpgradeKubeOneCluster(ctx context.Context,
-	externalCluster *kubermaticv1.ExternalCluster,
-	oldCluster *apiv2.ExternalCluster,
-	newCluster *apiv2.ExternalCluster,
-	externalClusterProvider provider.ExternalClusterProvider,
-	masterClient ctrlruntimeclient.Client,
-) (*apiv2.ExternalCluster, error) {
-	manifest := externalCluster.Spec.CloudSpec.KubeOne.ManifestReference
-
-	manifestSecret := &corev1.Secret{}
-	if err := masterClient.Get(ctx, types.NamespacedName{Namespace: manifest.Namespace, Name: manifest.Name}, manifestSecret); err != nil {
-		return nil, utilerrors.NewBadRequest(fmt.Sprintf("can not retrieve kubeone manifest secret: %v", err))
-	}
-	currentManifest := manifestSecret.Data[resources.KubeOneManifest]
-
-	cluster := &kubeonev1beta2.KubeOneCluster{}
-	if err := yaml.UnmarshalStrict(currentManifest, cluster); err != nil {
-		return nil, fmt.Errorf("failed to decode manifest secret data: %w", err)
-	}
-	upgradeVersion := newCluster.Spec.Version.Semver().String()
-	cluster.Versions = kubeonev1beta2.VersionConfig{
-		Kubernetes: upgradeVersion,
-	}
-
-	if oldCluster.Spec.ContainerRuntime == resources.ContainerRuntimeDocker {
-		cluster.ContainerRuntime.Containerd = nil
-		if upgradeVersion >= "1.24" {
-			return nil, utilerrors.NewBadRequest("container runtime is \"docker\". Support for docker will be removed with Kubernetes 1.24 release.")
-		} else if cluster.ContainerRuntime.Docker == nil {
-			cluster.ContainerRuntime.Docker = &kubeonev1beta2.ContainerRuntimeDocker{}
-		}
-	}
-
-	patchManifest, err := yaml.Marshal(cluster)
+	cluster := existingCluster.DeepCopy()
+	clusterToPatchJSON, err := json.Marshal(cluster)
```

Comment:
How about this naming?
```suggestion
	exisitingClusterJSON, err := json.Marshal(cluster)
```